fs/fs_test.go

```diff
@@ -100,11 +100,63 @@ func TestDirUsage(t *testing.T) {
 	fi, err := f.Stat()
 	as.NoError(err)
 	expectedSize := uint64(fi.Size())
-	size, err := fsInfo.GetDirUsage(dir, time.Minute)
+	size, err := fsInfo.GetDirDiskUsage(dir, time.Minute)
 	as.NoError(err)
 	as.True(expectedSize <= size, "expected dir size to be at-least %d; got size: %d", expectedSize, size)
 }
 
+//make sure that the timeout is actually being triggered (found this bug in PR#1489)
+func TestDirDiskUsageTimeout(t *testing.T) {
+	as := assert.New(t)
+	fsInfo, err := NewFsInfo(Context{})
+	as.NoError(err)
+	dir, err := ioutil.TempDir(os.TempDir(), "")
+	as.NoError(err)
+	defer os.RemoveAll(dir)
+	dataSize := 1024 * 10000 //1000 KB bigger to make sure it triggers tihe timeout
+	b := make([]byte, dataSize)
+	f, err := ioutil.TempFile(dir, "")
+	as.NoError(err)
+	as.NoError(ioutil.WriteFile(f.Name(), b, 0700))
+	_, err = fsInfo.GetDirDiskUsage(dir, time.Nanosecond)
+	as.Error(err)
+}
+
+func TestDirInodeUsage(t *testing.T) {
+	as := assert.New(t)
+	fsInfo, err := NewFsInfo(Context{})
+	as.NoError(err)
+	dir, err := ioutil.TempDir(os.TempDir(), "")
+	as.NoError(err)
+	defer os.RemoveAll(dir)
+	numFiles := 1000
+	for i := 0; i < numFiles; i++ {
+		_, err := ioutil.TempFile(dir, "")
+		as.NoError(err)
```

Comment:
I think you should use a require here, otherwise this could be a lot of error spam.